md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/AnalyticsAppAPI.java

```diff
@@ -0,0 +1,279 @@
+import com.wipro.ats.bdre.md.dao.UserRolesDAO;
+import com.wipro.ats.bdre.md.dao.jpa.*;
+import com.wipro.ats.bdre.md.dao.jpa.Process;
+import com.wipro.ats.bdre.md.dao.jpa.Properties;
+import com.wipro.ats.bdre.md.rest.RestWrapper;
+import com.wipro.ats.bdre.md.rest.util.Dao2TableUtil;
+import com.wipro.ats.bdre.md.rest.util.DateConverter;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+import java.util.*;
+
+/**
+ * Created by SU324335 on 4/13/2016.
+ */
+@Controller
+@RequestMapping("/analyticsapp")
+public class AnalyticsAppAPI {
+    private static final Logger LOGGER = Logger.getLogger(AnalyticsAppAPI.class);
+    @Autowired
+    private ProcessDAO processDAO;
+    @Autowired
+    private AnalyticsAppsDAO analyticsAppsDAO;
+    @Autowired
+    UserRolesDAO userRolesDAO;
+
+    @RequestMapping(value = {"/createjobs"}, method = RequestMethod.POST)
+
+    @ResponseBody
+    public RestWrapper createJob(HttpServletRequest request,@RequestParam Map<String, String> map, Principal principal) {
+        LOGGER.debug(" value of map is " + map.size());
+        String appImagePathRequest = request.getParameter("appimage");
+        LOGGER.info("Path from request "+appImagePathRequest);
+
+        String actualPath = request.getRealPath(appImagePathRequest);
+        LOGGER.info("actual Path "+actualPath);
+        RestWrapper restWrapper = null;
+
+        String processName = null;
+        String processDescription = null;
+        Integer busDomainId = null;
+
+
+
+        List<com.wipro.ats.bdre.md.dao.jpa.Properties> propertiesList = new ArrayList<Properties>();
+
+        for (String string : map.keySet()) {
+            com.wipro.ats.bdre.md.dao.jpa.Properties jpaProperties=new Properties();
+            LOGGER.info("String is" + string);
+            if (map.get(string) == null || ("").equals(map.get(string))) {
+                continue;
+            }
+            else if (string.startsWith("appproperties_processName")) {
+                LOGGER.debug("process_processName" + map.get(string));
+                processName = map.get(string);
+            }else if (string.startsWith("appproperties_processDescription")) {
+                LOGGER.debug("process_processDescription" + map.get(string));
+                processDescription = map.get(string);
+            }else if (string.startsWith("appproperties_busDomainId")) {
+                LOGGER.debug("appproperties_busDomainId" + map.get(string));
+                busDomainId = new Integer(map.get(string));
+            }
+            else if (string.startsWith("appproperties_industry")) {
+                jpaProperties = Dao2TableUtil.buildJPAProperties("analytics-app", "industry", map.get(string), "Industry Name");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "analytics-app" 7 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)